cps-drivers/aio/cpsaio.c

```diff
@@ -2367,7 +2369,7 @@ static long cpsaio_ioctl( struct file *filp, unsigned int cmd, unsigned long arg
 					}
 					spin_lock_irqsave(&dev->lock, flags);
 					
-					cps_common_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
+					contec_mcs341_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
 					DEBUG_CPSAIO_COMMAND(KERN_INFO"DIRECT INPUT:[%lx]=%x\n",(unsigned long)( dev->baseAddr + d_ioc.addr ), valw );
 					d_ioc.val = (unsigned long) valw;
```

Comment:
If d_ioc.val has the possibility of being updated by an interrupt, I feel the need to call spin_unlock_irqrestore after copy_to_user () on line 2378.

Reply:
I was understanding.
I change from spin_lock_irqsave function to spin_lock function in IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , I change from spin_unlock_irqrestore function to spin_unlock function.

Reply:
You said, "Do you need to spinlock before copy_from_user (and copy_to_user)?"
I made new issue #12.
I am researching a new branch.